backend/src/main/java/gov/cdc/usds/simplereport/service/DbOrgRoleClaimsService.java

```diff
@@ -0,0 +1,147 @@
+package gov.cdc.usds.simplereport.service;
+
+import gov.cdc.usds.simplereport.api.model.errors.MisconfiguredUserException;
+import gov.cdc.usds.simplereport.api.model.errors.NonexistentUserException;
+import gov.cdc.usds.simplereport.config.authorization.OrganizationRole;
+import gov.cdc.usds.simplereport.config.authorization.OrganizationRoleClaims;
+import gov.cdc.usds.simplereport.db.model.ApiUser;
+import gov.cdc.usds.simplereport.db.model.Facility;
+import gov.cdc.usds.simplereport.db.model.Organization;
+import gov.cdc.usds.simplereport.db.repository.ApiUserRepository;
+import gov.cdc.usds.simplereport.service.model.IdentitySupplier;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class DbOrgRoleClaimsService {
+  private final ApiUserRepository _userRepo;
+  private final IdentitySupplier _getCurrentUser;
+
+  /**
+   * Fetches the user by username and returns a list of OrganizationRoleClaims from the DB throws an
+   * exception if user does not have ONLY one org
+   *
+   * @param username - String of user email/username
+   * @return List of OrganizationRoleClaims from the DB
+   */
+  public List<OrganizationRoleClaims> getOrganizationRoleClaims(String username) {
+    try {
+      ApiUser user =
+          _userRepo.findByLoginEmail(username).orElseThrow(NonexistentUserException::new);
+      return List.of(getOrganizationRoleClaims(user));
+    } catch (NonexistentUserException | MisconfiguredUserException e) {
+      return new ArrayList<>();
+    }
+  }
+
+  public OrganizationRoleClaims getOrganizationRoleClaims(ApiUser user) {
+    Set<Organization> orgs = user.getOrganizations();
+    if (orgs.size() != 1) {
+      log.error("Misconfigured organizations in DB for User ID: {}", user.getInternalId());
+      throw new MisconfiguredUserException();
+    }
+    Set<OrganizationRole> roles = user.getRoles();
+    Set<UUID> facilityUUIDs =
+        user.getFacilities().stream().map(Facility::getInternalId).collect(Collectors.toSet());
+
+    String orgExternalId = orgs.stream().findFirst().get().getExternalId();
+    return new OrganizationRoleClaims(orgExternalId, facilityUUIDs, roles);
+  }
+
+  /**
+   * Compares two lists of OrganizationRoleClaims and checks if they are equal; If they are not
+   * equal, a message is logged with the affected User ID
+   *
+   * @param oktaClaims - List<OrganizationRoleClaims> from Okta
+   * @param dbClaims - List<OrganizationRoleClaims> from DB
+   * @return boolean
+   */
+  public boolean checkOrgRoleClaimsEquality(
```

Comment:
⚠️ Currently only using this in the tests but will use it later to introduce logging and alerting based on differing org roles from Okta vs DB